### ISSUE TITLE
Segmentation fail due to neg length vector

### ISSUE THREAD

**Issue by mpi-pebert** [lamortenera/epicseg#7]
I have a problem generating a segmentation for several samples at once (procedure as described in the manual: make the count matrices, normalize the count matrices and then do the segmentation) due to the following error:

```R
Error in .Call("kfoots_fillPosteriors", PACKAGE = "kfoots", coords, clusters,  : 
      negative length vectors are not allowed
    Calls: <Anonymous> ... do.call -> <Anonymous> -> initAlgo -> fillPosteriors -> .Call
    Execution halted
```

The rest of the stdout is as follows:
```R
loading epicseg
    [init] performing PCA on the count matrix 
    [init] splitting axes 
    [init] computing average seeds 
    [init] estimating r parameter 
    [init] computing KL divergences 
    [init] hierarchical clustering 
    [init] filling in posterior matrix
```

At first glance, the count matrices look ok - any idea where this is coming from?

**Comment by lamortenera** [comment]
Hi Peter,
yes, this is almost certainly due to very large vectors (matrices are stored as vectors). How many states, histone marks and bins do you have?

**Comment by mpi-pebert** [comment]
Hi Ale,
it's the usual: 200bp bins, 6 marks per sample, 9 samples and 18 states.
Best
Peter
Am 28.10.2015 5:35 nachm. schrieb lamortenera <notifications@github.com>:Hi Peter,
yes, this is almost certainly due to very large vectors (matrices are stored as vectors). How many states, histone marks and bins do you have?

**Comment by lamortenera** [comment]
I will have a look at it this week end! Thanks for reporting!
Am 28.10.2015 5:08 nachm. schrieb "Peter Ebert" <notifications@github.com>:

**Comment by lamortenera** [comment]
Ciao Peter,
the problem was indeed the large size of the dataset. the posterior matrix (which is the largest allocated vector) is 18(nstates)*9(ndatasets)* 15*10^6(nbins) long, which is a number that cannot be stored in a 32 bits integer. Luckily R has introduced the so called "long vectors" which should remove this issue. I had to do only very minor changes to the kfoots package for it to support long vectors (and I am not even sure that I needed to change anything). The problem was the Rcpp package, where the support for long vector has been introduced only very recently. 
In summary, try to do the following:
1. Install the latest version of the Rcpp package (unfortunately, not yet on CRAN):

    >library(devtools)
    >devtools::install_github("RcppCore/Rcpp")

2. re-install kfoots:

    >devtools:::install_github("lamortenera/kfoots")

Let me know if it works!

Ale

**Comment by mpi-pebert** [comment]
I had the time to work on this big analysis again; installing the current version of Rcpp and re-installing EpiCSeg/kfoots solved the problem. Ticket can be closed.
